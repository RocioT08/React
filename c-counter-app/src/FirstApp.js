import React, { Fragment } from "react";
import PropTypes from "prop-types";

const FirstApp = ({ personName, detail }) => {
  const greeting = `Hola Mundo: ${personName}`;
  const person = {
    name: "Juan",
    age: 14,
  };

  console.log({ personName });

  return (
    <Fragment>
      <h1>{greeting}</h1>
      <p className="jsonText">{JSON.stringify(person)}</p>
      <p className="detailText">{detail}</p>
    </Fragment>
  );
};

FirstApp.propTypes = {
  personName: PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
  detail: "This is a default detail in propTypes",
};

export default FirstApp;
