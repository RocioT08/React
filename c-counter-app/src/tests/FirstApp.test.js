import React from "react";
//import { render } from "@testing-library/react";
import { shallow } from "enzyme";
import FirstApp from "../FirstApp";
import "@testing-library/jest-dom";

describe("FirstApp", () => {
  // NOTA: ESTA PRUEBA SE COMENTA POR QUE YA NO VA A FUNCIONAR CON ENZYME YA QUE ESTA USA JEST
  // test("should return Hola Mundo: Juan Pablo", () => {
  //   const greeting = "Hola Mundo: Juan Pablo";
  //   const { getByText } = render(<FirstApp personName="Juan Pablo" />);
  //   expect(getByText(greeting)).toBeInTheDocument();
  // });

  test("should display <FirstApp/> correctly", () => {
    const greeting = "Hola Mundo: Juan Pablo";
    const wrapper = shallow(<FirstApp personName="Juan Pablo" />);
    //la respuesta sea igual a el snapshot
    expect(wrapper).toMatchSnapshot();
  });

  test("should display the detail sent as props", () => {
    const greeting = "Juan";
    const detail = "This is a default detail in propTypes";
    const wrapper = shallow(<FirstApp personName={greeting} detail={detail} />);

    //.text(), .html()
    //estoy buscando un elemento que tenga esa clase, tambien puedo mandar a buscar por tipo <p>
    const detailContentInSnapshot = wrapper.find(".detailText");

    console.log("<p> content = ", detailContentInSnapshot.html());
    expect(detailContentInSnapshot.text()).toEqual(detail);
  });
});
