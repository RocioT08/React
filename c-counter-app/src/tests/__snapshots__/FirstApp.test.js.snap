// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirstApp should display <FirstApp/> correctly 1`] = `
<Fragment>
  <h1>
    Hola Mundo: Juan Pablo
  </h1>
  <p
    className="jsonText"
  >
    {"name":"Juan","age":14}
  </p>
  <p
    className="detailText"
  >
    This is a default detail in propTypes
  </p>
</Fragment>
`;
